modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java

```diff
@@ -232,6 +233,34 @@ public void testAddSystemObjectRelationship() throws Exception {
 		_testSystemObjectRelationshipManyToMany();
 	}
 
+	@Test
+	public void testMandatoryRelationship() throws PortalException {
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				_objectDefinition1.getObjectDefinitionId(),
+				_objectDefinition2.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap("Able"), StringUtil.randomId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		ObjectField objectField = _objectFieldLocalService.updateRequired(
+			objectRelationship.getObjectFieldId2(), true);
+
+		Assert.assertTrue(objectField.isRequired());
+
+		_objectRelationshipLocalService.updateObjectRelationship(
+			objectRelationship.getObjectRelationshipId(),
+			objectRelationship.getParameterObjectFieldId(),
+			ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE,
+			objectRelationship.getLabelMap());
+
+		ObjectField newObjectField = _objectFieldLocalService.fetchObjectField(
```

Comment:
You don't need another variable for it.

Reply:
So, i understand that we need the new variable because the update relationship needs change the objectField to don´t mandatory. It make sense?

Reply:
Disregard message above! Now, i understand.